src/openrct2/world/TileElement.h

```diff
@@ -83,6 +83,7 @@ struct TileElementBase
 struct TileElement : public TileElementBase
 {
     uint8_t pad_04[4];
+    uint8_t pad_08[8];
```

Comment:
I think the padding should **only** be added to `TileElement` and not to any of the others. All of the other structs should 16 or less bytes. `TileElement` is the only one that needs to be `16`. That way we don't have to clutter the code base with pragmas. Its how we have done the sprites (kind of they use a big union).

Reply:
That will cause the assertions of the other tile element types to fail. Not an option.

Reply:
@duncanspumpkin While the others don't need to be 16 bytes, it is safer. For example is I were do `SurfaceElement elements[5];` I would probably want that to be 5 * 16 bytes.